### DISCUSSION TITLE
How to detect rejoin information

### DISCUSSION THREAD

**Discussion by agatah2333** [zwave-js/zwave-js#4464] [Q&A]
Hello developer, node-zwave-js is a perfect project using python to build connections with Zwave devices. I am trying to send some commands to my own devices. But sometimes, I find the device will restart. The device restart is like the device will be offline and rejoin with the dongle. Could anyone help me detect when the device is offline?

**Comment by AlCalzone** [comment]
> Could anyone help me detect when the device is offline?

Knowing the actual status of a device isn't really something the Z-Wave protocol was built to support.

node-zwave-js tries to make a **best effort guess** based on the communication with the node. If commands fail multiple times due to NoACK (no acknowledgement from the device), the device will be marked as dead (`node.status === NodeStatus.Dead`). If it later comes back alive, the status will be updated when either the node sends something, or a command (e.g. a ping) to the node succeeds.

How do you know that the device actually restarts though? Intermittent problems communicating with a device often come from a bad controller placement or a suboptimal mesh. To investigate that, I suggest the following:
1. read the intro to the troubleshooting guide https://zwave-js.github.io/node-zwave-js/#/troubleshooting/index
2. make sure you've followed the general tips here: https://zwave-js.github.io/node-zwave-js/#/troubleshooting/connectivity-issues?id=general-troubleshooting - a bad controller placement can easily cause what you're experiencing
3. Try doing a network heal, especially if you've moved the controller as part of step 2.
4. If that doesn't help, use zwavejs2mqtt (or any other application that has implemented the health checks already) to check the health of each node's route to the controller (node 1). You might find some don't have a strong link

**Reply by agatah2333** [reply]
Thanks for your reply! 
I tried to imitate the restart of the device, such as repeatedly plugging and unplugging my own smart plug device (Minoston smart plug). The Zwave-js log will show something like the attachment below when I plug the device.

<img width="911" alt="image" src="https://user-images.githubusercontent.com/90156813/163212812-56fec806-6963-411a-9b8e-6df0b161d0ba.png">

I wonder how I could trigger this frame? Does it mean the device rejoin?

**Reply by AlCalzone** [reply]
That's just a report telling you that configuration parameter 85 has value 5. Maybe your device sends that on startup, but it is definitely not a standard "hey I'm alive" message.

Also I'm not sure what you mean by "trigger this frame"?

**Reply by agatah2333** [reply]
Thanks for your helpful explanation!

 I think if the device could report the configuration on each startup time, this report configuration frame( sorry if I use the wrong word to describe), 0x010b00040006057006550105d4, could be regarded as a sign to figure out whether the device restart?

**Reply by AlCalzone** [reply]
I mean you could watch for the configuration parameter 85 changing/updating. I'm not convinced that this is a foolproof way to determine the device status though.
